[leezhengjing] iP

leezhengjing opened the pull request:
# BeeBot
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

BeeBot is a friendly bee that reminds you of your todos, deadlines and events!

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. Check out the repository [here!](https://github.com/leezhengjing/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
And it is FREE!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
If you Java programmer, you can use it to practice Java too. Here's the main method:

``` 
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

leezhengjing committed:
Completed Level-0

- Changed the name of chatbot to Bee
- Replaced ASCII art to Bee
- Added new print statements following the example code

leezhengjing committed:
Modified README
- Updated README and other files

leezhengjing committed:
Completed Level-1 Increment
- Created a while loop to scan user input
- Exit condition of "bye"
- Echos the user input

leezhengjing committed:
Completed Level-2 Increment
- Created a fixed size array to store input
- Added an else-if condition to return the stored inputs

leezhengjing committed:
Completed Level-3 Increment
- Created new classes Task and TaskList
- Moved some logic in the main loop to TaskList
- Added else-if statements to check for mark and unmark

leezhengjing committed:
Completed Level-4 Increment
- Added Todo, Deadline and Event classes that inherit Task
- Used the .substring() method to access the information in user input
- Used the .split() method to access the information if user creates
  deadline or event

leezhengjing committed:
Completed A-TextUiTesting
- Added a few sample lines into input.txt
- Created a EXPECTED.TXT file to match the input
- Ran the runtest.sh file and kept failing despite same actual and
  expected
- Passed the test after installing dos2unix

leezhengjing committed:
Completed Level-5 Increment
- Created BeeException class which extends exception
- Generates a bee ascii art text along with the error message in the
  case of incorrect outputs

leezhengjing committed:
Completed Level-6 Increment
- Added delete functionality in TaskList
- Refactored the TaskList code to use ArrayList<Task>
- Minor bug fixes for the error handling in Bee

leezhengjing committed:
Completed A-Enums Increment
- Added TaskClass and TaskAction enums to represent the elifs
- Refactored the code to createTask and updateTask methods

leezhengjing committed:
Implemented Level-7 Increment
- Changed if-else main block to switch statements for readability
- Added loadTasksFromFile(), saveTasksToFile() and parseTask()
- Program can now save the output to a file and read from there
- Added new constructors in the task classes to help parseTask()

leezhengjing committed:
Completed Level-8 Increment
- Changed Deadline class to store LocalDateTime for the deadline
- Updated the createTask and parseTask logic to read the new format

leezhengjing committed:
Completed A-MoreOOP Increment
- Created Parser, Storage and Ui classes
- Extracted large chunk of code to those classes,  making code look
  neater

leezhengjing committed:
Completed A-Packages Increment
- Created a bee package and moved all the class files into the package

leezhengjing committed:
Merge branch 'add-gradle-support'

leezhengjing committed:
Completed A-JUnit increment
- Added JUnit tests for Deadline and TaskList classes

leezhengjing committed:
Completed A-Jar Increment
- Modified build.gradle
- Created a jar file

leezhengjing committed:
Completed A-JavaDoc Increment
- Wrote header comments to all classes and public methods

leezhengjing committed:
Completed A-CodingStandard Increment
- Modified and adusted code to follow the SE-EDU Java Coding Standard

leezhengjing committed:
Completed Level-9 Increment
- Added finding tasks by keyword functionality

leezhengjing committed:
Merge branch 'branch-A-CodingStandard'

leezhengjing committed:
Merge branch 'branch-Level-9'

Song-Mengfei commented on src/main/java/bee/Storage.java:
```diff
@@ -0,0 +1,132 @@
+                }
+                break;
+        }
+    }
```
Perhaps you should abstract out the operations so that the this method is not so long and difficult to read?

Song-Mengfei commented on src/main/java/bee/TaskList.java:
```diff
@@ -0,0 +1,269 @@
+            default:
+                this.addTask(new Task(userInput));
+        }
+    }
```
Is this similar with the codes in Storage? Maybe a seperate class containing this method could make the code clear?

Song-Mengfei commented on src/main/java/bee/BeeException.java:
```diff
@@ -0,0 +1,31 @@
+                "(\")(_)-\"()))=-\n" +
+                "   (\\\\ " + super.getMessage();
+    }
+}
```
Very nice bee. I like your design!

Song-Mengfei reviewed (commented):
I like your design! Some codes are a little bit hard for me to read, but I can get your idea. Overall good job!

leezhengjing committed:
Implemented A-Checkstyle Increment
- Added checkstyle files and used checkstyle gradle
- Fixed switch and case auto indenting issue
- Fixed all checkstyle errors

leezhengjing committed:
Completed Level-10 Increment
- Added JavaFX GUI to the chatbot
- Changed most methods from void to return a string

PohSayKeong commented on src/main/java/bee/Bee.java:
```diff
@@ -0,0 +1,81 @@
+     *
+     * @param filePath The path to the data file for storing tasks.
+     */
+    public Bee(String filePath) {
```
might wanna consider if there is a need for another constructor here

PohSayKeong commented on src/main/java/bee/Bee.java:
```diff
@@ -0,0 +1,81 @@
+
+        while (isRunning) {
+            String userInput = scanner.nextLine();
+//            isRunning = Parser.parseUserCommand(userInput, tasks, storage, ui);
```
the indentation of this comment should follow the coding standards

PohSayKeong commented on src/main/java/bee/Bee.java:
```diff
@@ -0,0 +1,81 @@
+        scanner.close();
+    }
+
+    /**
+     * Parses the user's response and returns Bee's output
+     */
```
consider including the input parameters in the javadoc

PohSayKeong commented on src/main/java/bee/TaskList.java:
```diff
@@ -0,0 +1,262 @@
+            try {
+                int taskIndex = Integer.parseInt(splitInput[1]);
+                return this.setTaskDone(taskIndex);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                return Ui.returnErrorString(new BeeException("OOPS!! The task number cannot be empty."));
+            } catch (NumberFormatException e) {
+                return Ui.returnErrorString(new BeeException("OOPS!! You must have entered an invalid task number."));
+            }
```
these catch statements are repeated across the different cases, consider moving the try catch out?

PohSayKeong commented on src/test/java/bee/DeadlineTest.java:
```diff
@@ -0,0 +1,28 @@
+
+public class DeadlineTest {
+    @Test
+    public void initializeTest1() {
```
do look up the naming convention for test functions

PohSayKeong commented on src/main/java/bee/ui/Ui.java:
```diff
@@ -0,0 +1,120 @@
+                + " |    |  _// __ \\_/ __ \\\n"
+                + " |    |   \\  ___/\\  ___/\n"
+                + " |______  /\\___  >\\___  >\n"
+                + "        \\/     \\/     \\/\n";
```
this is cool!

PohSayKeong reviewed (commented):
great job on the work so far!

leezhengjing committed:
Complete A-Assertions Increment
Added assert statements to various classes

leezhengjing committed:
Merge pull request #1 from leezhengjing/branch-A-Assertions

Complete A-Assertions Increment

leezhengjing committed:
Complete A-CodeQuality Increment
Changed code to better fit the code quality topics, and running
checkstyle again on all the newly written code.

leezhengjing committed:
Merge pull request #2 from leezhengjing/branch-A-CodeQuality

Complete A-CodeQuality Increment

leezhengjing committed:
Complete C-FriendlierSyntax increment
- Added minimal aliases for user commands

leezhengjing committed:
Merge pull request #3 from leezhengjing/branch-C-FriendlierSyntax

Complete C-FriendlierSyntax increment

leezhengjing committed:
Completed A-UserGuide Increment
- Created the user guide for BeeBot
- Uploaded a representative screenshot of BeeBot

leezhengjing committed:
Completed A-UserGuide Increment
- Created the user guide for BeeBot
- Added the representative screenshot for BeeBot

leezhengjing committed:
Merge pull request #5 from leezhengjing/branch-A-UserGuide

Created user guide

leezhengjing committed:
Completed A-MoreTesting increment
- Added more unit tests

leezhengjing committed:
Merge pull request #6 from leezhengjing/branch-A-MoreTesting

Completed A-MoreTesting increment

leezhengjing committed:
Update Storage.java

Fixed a bug where it was not creating a new file

leezhengjing committed:
Completed A-BetterGui increment
- Added a css style sheet for the GUI
- Improved the look of the GUI

leezhengjing committed:
Merge pull request #7 from leezhengjing/branch-A-BetterGui

Completed A-BetterGui increment

leezhengjing committed:
Changed tasklist

Fixed a bug where test case was failing

leezhengjing committed:
Merge branch 'master' of https://github.com/leezhengjing/ip

leezhengjing committed:
Changed the readme and ui

leezhengjing committed:
Update README.md

leezhengjing committed:
Update README.md
